subprojects/platform-native/src/main/groovy/org/gradle/nativeplatform/test/plugins/NativeBinariesTestPlugin.java

```diff
@@ -97,7 +97,9 @@ void attachBinariesToCheckLifecycle(ModelMap<Task> tasks, final ModelMap<NativeT
                 @Override
                 public void execute(Task checkTask) {
                     for (NativeTestSuiteBinarySpec testBinary : binaries) {
-                        checkTask.dependsOn(testBinary.getTasks().getRun());
+                        if (testBinary.isBuildable()) {
+                            checkTask.dependsOn(testBinary.getTasks().getRun());
+                        }
```

Comment:
@melix, @lptr, @bamboo you may want to take a look at this wrt the changes in 44a603d518dd04a85514fae6f55630236daae2d0. Not sure whether we should be considering an `#isBuildable` guard there as well.